jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -255,88 +277,61 @@ public boolean isContainedIn(Resource r)
+            catch (Exception e)
             {
-                // If the toRealPath() call fails, then let
-                // the alias checking routines continue on
-                // to other techniques.
-                LOG.trace("IGNORED", ioe);
+                if (e instanceof IOException)
+                    LOG.trace("IGNORED", e);
+                else
+                    LOG.warn("bad alias ({} {}) for {}", e.getClass().getName(), e.getMessage(), path);
+                // Not possible to serve this resource.
+                //  - This resource doesn't exist.
+                //  - No access rights to this resource.
+                //  - Unable to read the file or directory.
+                //  - Navigation segments (eg: "foo/../test.txt") would go through something that doesn't exist, or not accessible.
+                //  - FileSystem doesn't support toRealPath.
+                alias = false;
+                return;
             }
-        }
-
-        if (!abs.isAbsolute())
-            abs = path.toAbsolutePath();
-
-        // Any normalization difference means it's an alias,
-        // and we don't want to bother further to follow
-        // symlinks as it's an alias anyway.
-        Path normal = path.normalize();
-        if (!isSameName(abs, normal))
-            return normal;
 
-        try
-        {
-            if (Files.isSymbolicLink(path))
-                return path.getParent().resolve(Files.readSymbolicLink(path));
-            if (Files.exists(path))
-            {
-                Path real = abs.toRealPath();
-                if (!isSameName(abs, real))
-                    return real;
-            }
-        }
-        catch (IOException e)
-        {
-            LOG.trace("IGNORED", e);
+            /* If the path and targetPath are the same, also check
+             * The as-requested URI as it will represent what was
+             * URI created this PathResource.
+             * e.g. the input of `resolve("aa./foo.txt")
+             * on windows would resolve the path, but the Path.toUri() would
+             * not always show this extension-less access.
+             * The as-requested URI will retain this extra '.' and be used
+             * to evaluate if the targetPath.toUri() is the same as the as-requested URI.
+             *
+             * // On Windows
+             *  PathResource resource         = PathResource("C:/temp");
+             *  PathResource child            = resource.resolve("aa./foo.txt");
+             *  child.exists()                == true
+             *  child.isAlias()               == true
+             *  child.toUri()                 == "file:///C:/temp/aa./foo.txt"
+             *  child.getPath().toUri()       == "file:///C:/temp/aa/foo.txt"
+             *  child.getTargetURI()          == "file:///C:/temp/aa/foo.txt"
+             */
+            alias = !isSameName(path, targetPath) || !Objects.equals(uri, toUri(targetPath));
```

Comment:
I like this. Similar to what I saw in @lachlan-roberts PR.  Makes it very clear what an alias is....  but I'm on fence if it is clearer as:

```suggestion
            alias = !(isSameName(path, targetPath) && Objects.equals(uri, toUri(targetPath)));
```

thoughts?

Reply:
At first blush looks fine.
The bad / broken / non-existent / no-permission flows could be problematic until we merge PR #8702 .

Here's a few example scenarios.

path | targetPath | uri-as-requested | uri-from-target-path | alias
---- | ---- | ---- | ---- | ----
`C:/temp/aa./foo.txt` | `C:/temp/aa/foo.txt` | `file:///C:/temp/aa./foo.txt` | `file:///C:/temp/aa./foo.txt` | true
`/tmp/foo-symlink` | `/tmp/bar.txt` | `file:///tmp/foo-symlink` | `file:///tmp/bar.txt` | true
`C:/temp/aa.txt` | `C:/temp/AA.txt` | `file:///C:/temp/aa.txt` | `file:///C:/temp/AA.txt` | true
`/tmp/bar-exists/../foo.txt` | `/tmp/foo.txt` | `file:///tmp/bar-exists/../foo.txt` | `file:///tmp/foo.txt` | true
`/tmp/doesnt-exist.txt` | null (does not exist) | `file:///tmp/doesnt-exist.txt` | null (does not exist) | false
`/tmp/doesnt-exist/../foo.txt` | null (intermediate does not exist) | `file:///tmp/doesnt-exist/../foo.txt` | null (intermediate does not exist) | false
`/var/protected/config.xml` | null (no permissions) | `file:///var/protected/config.xml` | null (no permission) | false
`/tmp/foo-symlink` | null (broken symlink, doesn't point to anything) | `file:///tmp/foo-symlink` | null (broken symlink, doesn't point to anything) | false
`C:/temp/cannot:be:referenced` | null (illegal filename) | `file:///C:/temp/cannot:be:referenced` | null (illegal filename) | false